components/com_contact/src/Model/ContactModel.php

```diff
@@ -365,26 +365,28 @@ protected function buildContactExtendedData($contact)
         }
 
         // Get the profile information for the linked user
-        $userModel = $this->bootComponent('com_users')->getMVCFactory()
-            ->createModel('User', 'Administrator', ['ignore_request' => true]);
-        $data = $userModel->getItem((int) $contact->user_id);
-
-        PluginHelper::importPlugin('user');
-
-        // Get the form.
-        Form::addFormPath(JPATH_SITE . '/components/com_users/forms');
-
-        $form = Form::getInstance('com_users.profile', 'profile');
-
-        // Trigger the form preparation event.
-        Factory::getApplication()->triggerEvent('onContentPrepareForm', [$form, $data]);
-
-        // Trigger the data preparation event.
-        Factory::getApplication()->triggerEvent('onContentPrepareData', ['com_users.profile', $data]);
-
-        // Load the data into the form after the plugins have operated.
-        $form->bind($data);
-        $contact->profile = $form;
+        if ((int) $contact->user_id > 0) {
+            $userModel = $this->bootComponent('com_users')->getMVCFactory()
+                ->createModel('User', 'Administrator', ['ignore_request' => true]);
+            $data = $userModel->getItem((int) $contact->user_id);
+    
+            PluginHelper::importPlugin('user');
+    
+            // Get the form.
+            Form::addFormPath(JPATH_SITE . '/components/com_users/forms');
+    
+            $form = Form::getInstance('com_users.profile', 'profile');
+    
+            // Trigger the form preparation event.
+            Factory::getApplication()->triggerEvent('onContentPrepareForm', [$form, $data]);
+    
+            // Trigger the data preparation event.
+            Factory::getApplication()->triggerEvent('onContentPrepareData', ['com_users.profile', $data]);
+    
+            // Load the data into the form after the plugins have operated.
```

Comment:
```suggestion
            $data = $userModel->getItem((int) $contact->user_id);

            PluginHelper::importPlugin('user');

            // Get the form.
            Form::addFormPath(JPATH_SITE . '/components/com_users/forms');

            $form = Form::getInstance('com_users.profile', 'profile');

            // Trigger the form preparation event.
            Factory::getApplication()->triggerEvent('onContentPrepareForm', [$form, $data]);

            // Trigger the data preparation event.
            Factory::getApplication()->triggerEvent('onContentPrepareData', ['com_users.profile', $data]);

            // Load the data into the form after the plugins have operated.
```
Fix PHPCS (space at the end of line) - you probably should switch off auto-indent in your editor.

Reply:
And maybe it would be better to check for `$contact->user_id == 0` and do an early return in that case so it does not need that much code in the `if` block?